### ISSUE TITLE
mcmc chain returning all NaNs

### ISSUE THREAD

**Issue by darthoctopus** [grd349/PBjam#158]
specifically, this part of mcmc.py called by `run_epsilon`:

```
        idx = self.sampler.acceptance_fraction < accept_lim
        nbad = np.shape(pos[idx, :])[0]

        if nbad > 0:
            flatchains = self.sampler.chain[~idx, :, :].reshape((-1, self.ndim))
```

`flatchains` is being populated by all NaNs for one of the stars I am trying (16 Cyg A).

To reproduce:

```
import numpy as np
from astropy import units as u, constants as c
import lightkurve as lk

kic = '12069424' #16 Cyg A

ν = np.linspace(1000, 3000, 1000000) * u.uHz
ps = np.load('ps.npy')
pg = lk.Periodogram(ν, ps)

star = pb.star(kic, pg, 
               [2200, 50], [150,50], teff=[5830, 11], bp_rp=[0.8201175, 0.1],
               store_chains=True, 
               nthreads=4, make_plots=True)
```
with attachment [ps.zip](https://github.com/grd349/PBjam/files/3555792/ps.zip)

**Comment by nielsenmb** [comment]
Ah the problem is that the prior sample is not currently populated with
stars on the main sequence, only red giants. So the KDE returns -inf for
numax > 300ish.

We've been meaning to put an offset on the KDE pdf, so that this doesn't
happen.

**Comment by darthoctopus** [comment]
I see. would it be reasonable to permit a user-supplied data set for the ε prior, or to specify a value of ε without calling run_epsilon?

**Comment by nielsenmb** [comment]
Personally I'd like to have the option to add specified values of epsilon
when using the star class. This was an option in an earlier version of
PBjam, but was since removed and I don't remember why.

Adding a user-supplied set of priors is a bit tricky, since the prior is
10-dimensional, so you'd have to come up with a list of credible values of
numax, dnu, epsilon, curvature etc. etc.

We're in the process of updating the prior at the moment and then we'll
extend it to the SG and MS stars. We already have a sample of roughly 400
from Serenelli et al. 2014 (I think) and the Legacy sample, and a few more,
but this isn't enough. This issue is that the KDE bandwidths are optimized
hyperparameters, and this optimization is dominated by RG stars in our
sample. This is due to a very tight correlation in numax, dnu, and epsilon
for RGs, which causes the KDE bandwidths to become very small. The MS and
SG targets on the other hand have a relatively large spread in epsilon for
each dnu, and so the KDE doesn't return credible values when you're trying
to sample in between where you have MS and SG targets in the prior sample.
Basically for RG stars the KDE is very nice and smooth, but for MS stars
it's very spiky with -inf in between stars.

I think Lucas has a list of K2 targets that might be useful for this, and
we're getting at least a small set of SG stars from TESS as well. But it
might require a somewhat more drastic reworking of how the KDE is
constructed.

**Comment by nielsenmb** [comment]
A fair amount of main sequence and sub giant targets are now in the prior sample. It's still a bit sparse, so you might still experience some targets suffering from not having enough samples to construct the KDE from (the reason you get NaNs).
